### PR TITLE
Run net-istio tests on KinD with non-default cluster suffix 

### PR DESCRIPTION
**Pull request by nak3**
This patch adds net-istio tests for KinD on github action.
The test also uses random cluster suffix.

Fixes https://github.com/knative-sandbox/net-istio/issues/343 https://github.com/knative-sandbox/net-istio/issues/344 

/cc @ZhiminXiang @JRBANCEL @tcnghia

### CODE REVIEW

**Comment by tcnghia** [comment]
Very nice! Thanks @nak3 !

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    - name: Create KinD Cluster
+      run: |
+        set -x
+
+        # KinD configuration.
+        cat > kind.yaml <<EOF
+        apiVersion: kind.x-k8s.io/v1alpha4
+        kind: Cluster
+        nodes:
+        - role: control-plane
```
Comment (JRBANCEL) [inline .github/workflows/kind-e2e.yaml:100]: Are two nodes enough to run the E2E tests? In `serving`, I had to add more nodes, see https://github.com/knative/serving/blob/master/.github/workflows/kind-e2e.yaml#L139

Comment (nak3) [reply .github/workflows/kind-e2e.yaml:100]: Yes, as far as I tested, the conformance test for `networking(net-istio)` does not use much resources like `serving`. We can add extra nodes if the tests start failing in the future.

**Review by tcnghia — APPROVED** [review]
/lgtm

**Pull request merged**